frontend/src/components/SaveResultButton.tsx

```diff
@@ -124,13 +124,15 @@ const SaveResultButton: React.FC<{ props: SimulationProps }> = ({ props }) => {
             ) : (
                 <Button
                     onClick={handleSave}
-                    disabled={!Boolean(selectedProjectId && simulationName) && !isSimulationSaving}
+                    disabled={
+                        !Boolean(selectedProjectId && simulationName && simulationName.trim()) && !isSimulationSaving
+                    }
```

Comment:
Although I think the entire 122 to 137 could have been refactored out of this function (as it is nested within the return statements, doesn't look very nice..)

A suggestion though, how about setting for instance:

```ts
const saveAble = !Boolean(selectedProjectId && simulationName && simulationName.trim()) && 
!isSimulationSaving;
return (
....
disabled={saveAble}
```
to reduce code complexity. Perhaps also have a comment as to what why we validate it like this, if you think it makes sense